bbc/simorgh
# [WSTEAM1-53 + WSTEAM1-54] - Optimo articles ad support spike

**Overall change:**
Adds advert rendering logic and components to Optimo articles on **Test only** for now. Leaderboard adverts will appear above the article content at full-width, and MPU adverts will appear within the article content after the 4th instance of a paragraph.

These will only appear on Test until we have the appropriate metadata being returned from Ares for Optimo articles to determine whether or not to show ads.

No UX / UI changes have been made in this PR. This is purely to integrate ads into Optimo articles. Once designs are complete, any UX / UI changes will be made in a separate PR.

**Code changes:**
- Updates the `<ArticlePage />` component to include the ad components and script tags
- Updates the article `getInitialData` function to insert the MPU block after the 4th instance of a paragraph 
- Creates an `addMpuBlock` function, similar to what we have in CPS to determine where to insert the MPU block

**Testing locally**
- Add the following query parameters to the article page URL `?site=test&ads-debug=true`
- Install the Chrome extension [ModHeader](https://chrome.google.com/webstore/detail/modheader/idgpnmonknjnojddfkpgkljpfnnfcklj?hl=en) and add the header `bbc-adverts : true`
- These 2 steps should enable viewing of adverts locally

---

- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
